lib/puppet/resource_api/simple_provider.rb

```diff
@@ -7,7 +7,12 @@ module Puppet::ResourceApi
   class SimpleProvider
     def set(context, changes)
       changes.each do |name, change|
-        is = change.key?(:is) ? change[:is] : (get(context) || []).find { |r| r[:name] == name }
+        is = if context.feature_support?('simple_get_filter')
+               change.key?(:is) ? change[:is] : (get(context, [name]) || []).first
```

Comment:
according to the spec, `get(context, [name])`  is allowed to return more than one resource, and the first is not necessarily the one we were asking for. This still needs the same `.find { |r| r[:name] == name }` as below.